acceptance/analyzer_test.go

```diff
@@ -395,343 +575,150 @@ func testAnalyzer(t *testing.T, when spec.G, it spec.S) {
+								),
+							)
 
-				it.After(func() {
-					h.DockerImageRemove(t, authRegAppImage)
-				})
+							h.AssertMatch(t, output, "2222 3333 .+ \\.")
+							h.AssertMatch(t, output, "2222 3333 .+ committed")
+							h.AssertMatch(t, output, "2222 3333 .+ staging")
+						})
+					})
 
-				it("restores app metadata", func() {
-					output := h.DockerRunAndCopy(t,
-						containerName,
-						copyDir,
-						ctrPath("/layers"),
-						analyzeImage,
-						h.WithFlags(
-							"--network", registryNetwork,
-							"--env", "CNB_PLATFORM_API="+latestPlatformAPI,
-						),
-						h.WithArgs(
-							ctrPath(analyzerPath),
-							noAuthRegAppImage,
-						),
-					)
+					when("the provided cache directory is writeable by the CNB user's group", func() {
+						it("doesn't chown the directory", func() {
+							h.SkipIf(t, runtime.GOOS == "windows", "Not relevant on Windows")
+							h.SkipIf(t, api.MustParse(apiString).Compare(api.MustParse("0.7")) >= 0, "Platform API >= 0.7")
 
-					assertLogsAndRestoresAppMetadata(t, copyDir, output)
-				})
+							cacheVolume := h.SeedDockerVolume(t, cacheFixtureDir)
+							defer h.DockerVolumeRemove(t, cacheVolume)
 
-				when("skip layers is provided", func() {
-					it("writes analyzed.toml and does not write buildpack layer metadata", func() {
-						output := h.DockerRunAndCopy(t,
-							containerName,
-							copyDir,
-							ctrPath("/layers"),
-							analyzeImage,
-							h.WithFlags(
-								"--network", registryNetwork,
-								"--env", "CNB_PLATFORM_API="+latestPlatformAPI,
-							),
-							h.WithArgs(
-								ctrPath(analyzerPath),
-								"-skip-layers",
-								noAuthRegAppImage,
-							),
-						)
+							output := h.DockerRun(t,
+								analyzeImage,
+								h.WithFlags(append(
+									dockerSocketMount,
+									"--volume", cacheVolume+":/cache",
+									"--env", "CNB_PLATFORM_API="+apiString,
+								)...),
+								h.WithBash(
+									fmt.Sprintf("chown -R 9999:3333 /cache; chmod -R 775 /cache; %s -daemon -cache-dir /cache some-image; ls -alR /cache", analyzerPath),
+								),
+							)
 
-						assertAnalyzedMetadata(t, filepath.Join(copyDir, "layers", "analyzed.toml"))
-						assertWritesStoreTomlOnly(t, copyDir, output)
+							h.AssertMatch(t, output, "9999 3333 .+ \\.")
+							h.AssertMatch(t, output, "9999 3333 .+ committed")
+							h.AssertMatch(t, output, "2222 3333 .+ staging")
+						})
 					})
 				})
 			})
 		})
 
-		when("cache is provided", func() {
-			when("cache image case", func() {
+		when("registry case", func() {
+			it("writes analyzed.toml", func() {
+				h.DockerRunAndCopy(t,
+					containerName,
+					copyDir,
+					ctrPath("/layers/analyzed.toml"),
+					analyzeImage,
+					h.WithFlags(
+						"--env", "CNB_PLATFORM_API="+apiString,
+					),
+					h.WithArgs(
+						ctrPath(analyzerPath),
+						"some-image",
```

Comment:
Interesting not to see a `SkipIf` here - so for newer platform API, the image argument is just ignored?

Reply:
That is a good point. I would have expected a failure, but maybe all our binaries don't fail with additional arguments by default. Maybe we should fail?

Reply:
Looking at https://github.com/buildpacks/lifecycle/blob/1665484df8b2158c9b34a3c22b5156aacfc9f055/cmd/lifecycle/analyzer.go#L76-L84 it looks like `args[0]` is just ignored. Probably failing makes sense to be consistent with what we do in other similar scenarios.

Reply:
Is this still a thing?

Reply:
Added in 69189d7